Unix/tests/cli/test_cli.cpp

```diff
@@ -339,10 +343,45 @@ static int StopServerSudo()
 #else
     if (!skipTest)
     {
-        std::stringstream cmd;
-        cmd << "sudo kill -15 " << (int)serverProcess.reserved;
-        if (system(cmd.str().c_str()) == -1)
+        uint args = 0;
+        const char* argv[MAX_SERVER_ARGS];
+        std::stringstream pidStr;
+        int pid;
+        Process killProcess;
+        int status;
+        int numWaits = 0;
+
+        if (PIDFile_Read(&pid) != 0)
             return -1;
+
+        pidStr << pid;
+
+        argv[args++] = sudoPath;
+        argv[args++] = "-E";
```

Comment:
The -E qualifier doesn't exist on non-Linux platforms, so this won't work.

What we do in SCX is something like this:

Generate a temporary file with shell-sourceable lines to set env for specific items

Then we execute a command script with something like:

. env-script
command-to-execute

The env script will contain something like:

MYVAR1=value-of-MYVAR1
MYVAR2=value-of-MYVAR2

You can write this sort of thing from shell with something like:

echo "MYVAR1=$MYVAR1" >> env-script

Let me know if you have questions on this. But as is, this will definitely fail on non-Linux platforms. A PBUILD should show this.